api/src/main/java/org/ehcache/spi/serialization/Stateful.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.spi.serialization;
+
+import org.ehcache.spi.persistence.StateRepository;
+
+/**
+ * Serializers implementing this interface can have persistent state.
+ * The state will be persisted in the persistent tier of the cache.
+ *
+ */
+public interface Stateful {
```

Comment:
Let's have this named `StatefulSerializer` and `extends `Serializer` because there is no plan of having that contract exposed / honoured anywhere else.